### DISCUSSION TITLE
Set the width of a rendered gantt diagram

### DISCUSSION THREAD

**Discussion by white-gecko** [mermaid-js/mermaid#4188] [Q&A]
If I render a gantt diagram with mermaid or kroki it has different widths. How can I specify the width of the rendered image?

![](https://mermaid.ink/img/pako:eNqVVE1v2zAM_SuEgAIt4AT-iNPEt6BBhwHrLutlgw9VLSbRakuBJa_Jiv730VKc2ks6rDoYsun3yPdE8YUVWiDL2Jora3MFtAS3eKvriluA77RGd3ej5dLHrLQlgl8LIaRaw6fF1_t7EJKva17BqlGFlVrxUto9WA0VEpMUHo67omwEmnb_jPiEShgfubiAy4cu_AC8KHBrDZgtFnIlC1eUAal6FcHKFRlQbG9Ar8Bu0LHCZc5Mo-hzzq5A15CzYzIWwGNjQWnrf9e1OIRblpyNr3LlKzLohMCi2_nPN7ralmhRgOXmCXorE1ph4BxEEwUQh9FkFEajcBoctzNPsiDCX3jC0JJwFwocSdwjmQeQHFy8bWxTn0cfd4ROAuAri_WBKT1Fx_9AT3roxKOHvtzU0sqCl47JnDWHjqs1uej-LKVCyNpXOjRnVt-jycaTfG45KlQWtrw2VAFXAn5Ko1Wv1D7Jsc7W9IPKmxqpY4CahrqIGqXjGuj1JJ3j5-wlBcPqh9D0vWw1NRvWg3xZVxrdm97FgLMri47F9O8TFwJPEVklS8o8MDROAwhPzmypi6Z1lr_18xINaXlEcBMAzJ6iu_eakkdDt5OeJI_v5gAJFFhpsn2NHYkD8gigLXPzhuR0Gzct5ztYuljFMbHHT2abv7V94cYOr-r_SfNyXIYPyfmYBFcwC9jh1GnivrTonBGswpxltBW44k1paUb50LMUduNDcbjd5SxXr0TBG6u_7VXBMls3GLBm2w7Hpc_MshUvDX1FIa2u7_x0d0M-YFuufmj99g-9s-yF7RiJicdRnCZJOp3TY57MArZnWRSG43mYJlGUziaT63g6fQ3Yb0cRjmfXYRqGyTSOkvlsOrt-_QP0eNrX?type=png)

![](https://kroki.io/mermaid/svg/eNqVU01v2zAMvfdXEAYKtIBTxPkYOt-CBB0GrDv1sltVibbV2lIg0Vvz70fJtWs3ybDyJIt-j49PZCkM0QVwKEF4Z10jCOAXx-z-frbbxRRpqhG62CilTQnfNj8fHkBpUTrRQNEaSdoaUWs6AFlokIm0imh8lXWr0IfzH8QXNMrHxOUlXD322UcQUuKePPg9Sl1oGRV50GYkB4qoMOXcwYMtgCqMpHCV-NbwbXIN1kHSF0pSeGoJjKXuV-tUlwwEyc31RVTiMcqHTX-Kt1vb7GskVEDCv8AocmUNptE19FkKi3m2ms2z2fxLOhxvI8eG6X7jEUHgEDGVRo7FiONrCsvOubuWWncaPJwYvExBFITujWh9BF78A7wagZcRPHFk6zRpKerI40_Zwu8TnJX9j7U2CHn45FeKNo3dWVWR43ugaNAQ7IXzXF0YBc_ac8V3mWOOQWNwu2tw65AHBHhGeGh4LnqqSasdR2_1CV9Z_lT6FLk-U8vxcKGbVMvfdPGGjFYATkae9UrGiyOUwmNA3uia606sXKxTmH94qp2VbbBUDAO8Q89dPCGUYcvBHzj5em4MRTZ1efneTQfvl517U9hYtrvEniPiRAYQJFYDUPDeVYHxDJT3SA5lO_jqtpq29UN4mqzl_3XVdRLpP9PJp9QHrX8BWKl92w)

**Comment by betterthanever2** [comment]
100 upvotes. 
It would be great to manipulate the width of the chart, possibly with a horizontal scrollbar for long periods

**Reply by white-gecko** [reply]
A scrollbar would not be part of mermaid-js but of the integration.

**Reply by betterthanever2** [reply]
It's hardly possible in any integration, if you can't control the width

**Comment by imysl0** [comment]
https://github.com/yuzutech/kroki/pull/1413 It may be useful, but I am still confused about how to use it

**Comment by Piratepie** [comment]
After doing some digging I found a way to set the width of any chart using the config.

So the config of the gantt diagram is declared here:

https://github.com/mermaid-js/mermaid/blob/31a287b7b0a45d1d96d0c781058b2b0b08c8089e/packages/mermaid/src/config.type.ts#L1207

It doesn't have any implementation for max width or height like the xyChart has, but it does extend the `BaseDiagramConfig`. 

https://github.com/mermaid-js/mermaid/blob/31a287b7b0a45d1d96d0c781058b2b0b08c8089e/packages/mermaid/src/config.type.ts#L190-L199

And here there's a `useWidth` that takes in a number which sets the width of the diagram.

So knowing that I made a config for the default gantt chart example which looks like this:

```yaml
config:
    gantt:
        useWidth: 100
```

[This seems to be working in the editor](https://mermaid.live/edit#pako:eNpdUU1rwzAM_StC56TY-XR8K5Tt1FMHY8MXEzutWWOXxIF1pf99dpJStofB0tN7FpZu2DqlkWOapsK2znbmyIWFgKO03q9xxDTqd6P8iUNJiLCzQdhZtYi88WcNW3iNFOyMPA6yX0pKev3ihl56gI-AdL9Pd7ulNurWG2fhsNwLuQUvxy94gkuaQEZokRIaTgI5UavUOn_Sw9PAZedDLilAtKi_bVb5Qr5Fj7GxuLZ5tKBZdNNsdcv_TYBDVihMsNfhW0aFEd6iVGDQ9VogD2HnBj16gcLeg1JO3h2utkXuh0knOF3iWNY5Ie_keQzsRdpP5_qHKKTIb_iNvKw2DaMVqSllJGc5TfCKnFbFhjCa51WWlU1BSHlP8Gd-gGzqIiOENawgjLGmrhPUyng37Jedz6u__wK89ZPf) but doesn't here on GitHub as you can see below. 

So I'm not sure if this will work for you, but trying never hurt anyone :p

```mermaid
---
config:
    gantt:
        useWidth: 500
---

gantt
    title A Gantt Diagram
    dateFormat  YYYY-MM-DD
    section Section
    A task           :a1, 2014-01-01, 30d
    Another task     :after a1  , 20d
    section Another
    Task in sec      :2014-01-12  , 12d
    another task      : 24d
```

**Reply by betterthanever2** [reply]
This is a great insight, thank you!

**Reply by imysl0** [reply]
It works!666~
